mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -2605,7 +2612,8 @@ void AsmPrinter::Impl::printOptionalAttrDict(ArrayRef<NamedAttribute> attrs,
 
     // Otherwise, print them all out in braces.
     interleave(
-        filteredAttrs, [&](NamedAttribute attr) { printNamedAttribute(attr); },
+        filteredAttrs,
+        [&](NamedAttribute attr) { printNamedAttribute(attr, separator); },
```

Comment:
For the inner separator, don't you need to add two extra spaces to make them visually indented?

Reply:
I initially implemented this. However, a modification on seperator here applies for every dictionary attribute (that is why seperation information passed from `AsmPrinter::Impl::printOptionalAttrDict` explicitly, while other calls to `AsmPrinter::Impl::printAttributeImpl` are not changed and using the default value for `seperator` argument)
 Consequently, we get the following IR if we add extra spaces here:
![image](https://github.com/Xilinx/llvm-project/assets/45471178/89fc9c6a-7099-4c65-a8d3-1f2610be2592)

Reply:
Right, you only want to add two extra spaces here if the 
```
if (printerFlags.getNewlineAfterAttrLimit() &&
        attrs.size() > *printerFlags.getNewlineAfterAttrLimit()) {
```
branch was taken above

Reply:
Could you please add a test case to show the capabilities (e.g. in mlir/test/IR/mlir-newline-after-attr.mlir)?